lib/battle/handler/BattlefieldHandler.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * BattlefieldHandler.cpp, part of VCMI engine
+ *
+ * Authors: listed in file AUTHORS in main folder
+ *
+ * License: GNU General Public License v2.0 or later
+ * Full text of license available in license.txt file, in main folder
+ *
+ */
+
+#include "StdInc.h"
+#include "BattlefieldHandler.h"
+#include "CModHandler.h"
+#include "battle/obstacle/ObstacleJson.h"
+
+BattlefieldHandler::BattlefieldHandler()
+{
+	const JsonNode obstacleConfig(JsonNode(ResourceID("config/obstacles.json")));
+
```

Comment:
* why handler for obstacle types called BattlefieldHandler? Battlefield type is another thing (see `BFieldType`).
* Handler should not access .json files directly. `config/obstacles.json` link should be in `gameConfig.json` (aka mod.json for Core mod)

Reply:
When I name it I hope that everything that is for arrangement of things and graphics on the battlefield, will be handled by this in future: siege, placement of units, graphics for battlefield etc. will go here. It's needs now a little refactor that seperate loading obstacles to another class, at all I was more worried about understand how quickly force newly handler to work to not waste too much time.

Reply:
The problem is that ModHandler do not designed to support one handler for different entity types. If you plan to make Battlefield type also configurable it will also need separate handler class.

Reply:
Maybe it was too far-reaching plan, BFieldType everything complicate, I will think about it a little more, I always change that name. However for future writing in mod.json 20 different jsons supported by 20 handlers in mod.json that set the one battle doesn't sound too good. It would be better to have one json that can set whole battle.